scalafix-cli/src/main/scala/scalafix/internal/v1/Args.scala

```diff
@@ -242,9 +247,28 @@ case class Args(
     RuleDecoder.decoder(ruleDecoderSettings.withConfig(scalafixConfig))
   }
 
+  // In running with onCompile flag, look for settings in onCompile block first, and fallback to standard settings.
+  //
+  // Please watch out that its merge scope is shallow. For example,
+  //   DisableSyntax.noVars = true
+  // will be overridden by
+  //   onCompile = {
+  //     DisableSyntax.noThrows = true
+  //   }
```

Comment:
I'm wondering what is the best... 🤔 
Or should we allow to override `DisableSyntax.noThrows` only,
 and keep `DisableSyntax.noVars = true` or other DisableSyntax settings?

Your feedback is much appreciated.

Reply:
I thought merging-shallow strategy is simpler and clearer, since users can understand what is going to run without merging Confs one by one in their mind.

However, I'd imagine several concrete use cases -- for example, users just want to turn `OrganizeImports.removeUnused` into `false` on compile. With merging-shallow strategy, they might face severe conf duplications especially in `OrganizeImports.groups` (or it can be generalized as `Conf.Lst`s), which are not pleasant.

So I have changed my mind and implementation with 342f74d . 
Now Confs are merged deeply, so that `onCompile` Conf.Obj can work just as a patch against original.

I'd still want to hear your opinions about this.